src/test/java/io/jenkins/plugins/analysis/warnings/RecorderITest.java

```diff
@@ -0,0 +1,218 @@
+package io.jenkins.plugins.analysis.warnings;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+import com.gargoylesoftware.htmlunit.html.DomElement;
+import com.gargoylesoftware.htmlunit.html.HtmlForm;
+import com.gargoylesoftware.htmlunit.html.HtmlFormUtil;
+import com.gargoylesoftware.htmlunit.html.HtmlNumberInput;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+
+import hudson.model.FreeStyleProject;
+import hudson.model.HealthReport;
+import hudson.model.Project;
+import hudson.model.Result;
+
+import io.jenkins.plugins.analysis.core.model.AnalysisResult;
+import io.jenkins.plugins.analysis.core.steps.IssuesRecorder;
+import io.jenkins.plugins.analysis.core.testutil.IntegrationTestWithJenkinsPerSuite;
+
+import static org.assertj.core.api.Assertions.*;
+
+/**
+ * Verifies the info page of the Java parser and the health report configuration via Web-Gui.
+ *
+ * @author Raphael Furch
+ */
+
+@RunWith(Parameterized.class)
+public class RecorderITest extends IntegrationTestWithJenkinsPerSuite {
+
+    private static final int HEALTHY_THRESHOLD = 1;
+    private static final int UNHEALTHY_THRESHOLD = 9;
+    private static final String INFORMATION_MESSAGE = "Enabling health report (Healthy=1, Unhealthy=9, Minimum Severity=LOW)";
+    /**
+     * Parameters under test.
+     * @return parameter list.
+     */
+    @Parameters
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] {
+                {"javac_0_warnings.txt",   0, 100, true},
+                {"javac_1_warnings.txt",   1,  90, true},
+                {"javac_9_warnings.txt",   9,  10, true},
+                {"javac_10_warnings.txt", 10,   0, true},
+                {"javac_0_warnings.txt",   0,  -1, false},
+                {"javac_1_warnings.txt",   1,  -1, false},
+                {"javac_9_warnings.txt",   9,  -1, false},
+                {"javac_10_warnings.txt", 10,  -1, false},
+        });
+    }
+
+    /**
+     * File with warnings.
+     */
+    @SuppressWarnings("checkstyle:visibilitymodifier")
+    @Parameter
+    public /* NOT private */ String javacSrcFile;
```

Comment:
Für was oder wen ist die Kennzeichnung `NOT private` gedacht?

Reply:
Wenn man die Variablen "private" setzt, funktioniert die Variablen-Injection nicht mehr.